src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -0,0 +1,1181 @@
+                var foundNode = node.AstIdentifierBase.Contains(((GeometryModel3D)e.OriginalSource).Name);
+                if (!foundNode) continue;
+                DynamoSelection.Instance.ClearSelection();
+                viewModel.Model.AddToSelection(node);
+            }
+        }
+
+        private static MeshGeometry3D DrawTestMesh()
+        {
+            var b1 = new MeshBuilder();
+            for (var x = 0; x < 4; x++)
+            {
+                for (var y = 0; y < 4; y++)
+                {
+                    for (var z = 0; z < 4; z++)
+                    {
+                        b1.AddBox(new Vector3(x, y, z), 0.5, 0.5, 0.5, BoxFaces.All);
+                    }
+                }
+            }
+            var mesh = b1.ToMeshGeometry3D();
+
+            mesh.Colors = new Color4Collection();
+            foreach (var v in mesh.Positions)
+            {
+                mesh.Colors.Add(new Color4(1f, 0f, 0f, 1f));
+            }
+
+            return mesh;
+        }
+
+        /// <summary>
+        /// This method attempts to maximize the near clip plane in order to 
+        /// achiever higher z-buffer precision.
+        /// </summary>
+        private void UpdateNearClipPlaneForSceneBounds(Rect3D sceneBounds)
+        {
+            // http: //www.sjbaker.org/steve/omniv/love_your_z_buffer.html
+            var maxDim = Math.Max(Math.Max(sceneBounds.SizeX, sceneBounds.Y), sceneBounds.SizeZ);
+            Camera.NearPlaneDistance = Math.Max(CalculateNearClipPlane(maxDim), 0.1);
+        }
+
+        internal override void ExportToSTL(string path, string modelName)
+        {
+            var geoms = SceneItems.Where(i => i is DynamoGeometryModel3D).
+                Cast<DynamoGeometryModel3D>();
+
+            using (TextWriter tw = new StreamWriter(path))
+            {
+                tw.WriteLine("solid {0}", model.CurrentWorkspace.Name);
+                foreach (var g in geoms)
+                {
+                    var n = ((MeshGeometry3D) g.Geometry).Normals.ToList();
+                    var t = ((MeshGeometry3D)g.Geometry).Triangles.ToList();
+
+                    for (var i = 0; i < t.Count(); i ++)
+                    {
+                        var nCount = i*3;
+                        tw.WriteLine("\tfacet normal {0} {1} {2}", n[nCount].X, n[nCount].Y, n[nCount].Z);
+                        tw.WriteLine("\t\touter loop");
+                        tw.WriteLine("\t\t\tvertex {0} {1} {2}", t[i].P0.X, t[i].P0.Y, t[i].P0.Z);
+                        tw.WriteLine("\t\t\tvertex {0} {1} {2}", t[i].P1.X, t[i].P1.Y, t[i].P1.Z);
+                        tw.WriteLine("\t\t\tvertex {0} {1} {2}", t[i].P2.X, t[i].P2.Y, t[i].P2.Z);
+                        tw.WriteLine("\t\tendloop");
+                        tw.WriteLine("\tendfacet");
+                    }
+                }
+                tw.WriteLine("endsolid {0}", model.CurrentWorkspace.Name);
+            }
+        }
+
+        #endregion
+    }
```

Comment:
Maybe you can give me some ideas of where you want me to focus in this file? I think a lot of this is just moved around...